src/main/java/com/github/princesslana/eriscasper/gateway/Gateway.java

```diff
@@ -87,19 +87,19 @@ private void sequenceNumberSeen(Optional<SequenceNumber> seq) {
     seq.ifPresent(sid -> lastSeenSequenceNumber = Optional.of(sid));
   }
 
-  @SuppressWarnings("unchecked")
+  private void warnOnClosing(RxWebSocketEvent evt) {
+    if (evt instanceof RxWebSocketEvent.Closing || evt instanceof RxWebSocketEvent.Closed) {
+      LOG.warn("Websocket closing: {}", evt);
```

Comment:
I have found information like the endpoint address (probably not relevant for the main websocket) and the shard id logged together with close events (code and reason) very helpful in some situations in the past (discord devs usually also require this information, when reporting rare:tm: bugs). Not sure how much of that information is available in the current Gateway implementation here; if it's available, it's worth adding imo.

Reply:
I think debugging and finding ways to show information for bugs should be done later when we're producing 1.0.0, not now, I don't think it's necessary to provide general debugging for users when the lib isn't even in a stable version yet.

Reply:
There's also probably going to be a lot of work done on the websocket still. Not worth providing user driven stuff for items we're just going to change. Me and Lana personally should have enough information when debugging or have the ability to get the information from a debugging process and reproduction of bugs when testing. Also our sharding isn't quite finished yet or in a production stage. It might be useful later on but there's a big TODO stamp in this section imo. It's not quite in a stage which provides a bunch of information (probably will be done in the documentation stages as well)

Reply:
The reason and code where already there. Have updated to include url and shard. Logged twice 'cos we log on closing and closed events.

```
[OkHttp https://gateway.discord.gg/...] WARN com.github.princesslana.eriscasper.gateway.Gateway - Websocket closing: url=wss://gateway.discord.gg, shard=Optional.empty, event=Closing{webSocket=okhttp3.internal.ws.RealWebSocket@10d5381, code=4004, reason=Authentication failed.}
[OkHttp WebSocket https://gateway.discord.gg/...] WARN com.github.princesslana.eriscasper.gateway.Gateway - Websocket closing: url=wss://gateway.discord.gg, shard=Optional.empty, event=Closed{webSocket=okhttp3.internal.ws.RealWebSocket@10d5381, code=4004, reason=Authentication failed.}
```

Reply:
That too ^

Reply:
I think there's still more we could do with providing all the information possible to everywhere in every case of exception. I'll look around of updates for information being passed. But imo that would be done when we start expanding the exception list for ErisCasper to start capturing specific cases.